torch/csrc/utils/auto_gpu.h

```diff
@@ -29,41 +30,31 @@ struct AutoGPU {
   }
 
   ~AutoGPU() {
-#ifdef USE_CUDA
-    if (original_device != -1) {
-      cudaSetDevice(original_device);
-    }
-#endif
+    #ifdef USE_CUDA
+      // Resets the original device (if necessary)
+      if (original_device != -1) {
+        int cur_device = -1;
+        THCudaCheck(cudaGetDevice(&cur_device));
+        if (cur_device != original_device) THCudaCheck(cudaSetDevice(original_device));
+      }
+    #endif
   }
 
-  inline void setDevice(int device) {
+    inline void setDevice(int device) {
 #ifdef USE_CUDA
     if (device == -1) {
       return;
     }
     if (original_device == -1) {
-      cudaCheck(cudaGetDevice(&original_device));
+      THCudaCheck(cudaGetDevice(&original_device));
       if (device != original_device) {
-        cudaCheck(cudaSetDevice(device));
+        THCudaCheck(cudaSetDevice(device));
       }
     } else {
-      cudaCheck(cudaSetDevice(device));
+      THCudaCheck(cudaSetDevice(device));
```

Comment:
I think it's better to use `cudaCheck` here, since we're generally moving away from `TH`

Reply:
No worries; sticking to master version on this while thing (per above).